nixos/modules/services/scheduling/scx.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.scx;
+in
+{
+  options.services.scx = {
+    enable = lib.mkEnableOption null // {
+      description = ''
+        Whether to enable SCX service, a scheduler daemon with wide
+        variety of scheduling algorithms, that can be used to improve system performance.
+
+        ::: {.note}
+        Requires a kernel with the Sched-ext feature and built with appropiate options.
+        Generally, 6.10 and later kernels are supported. See the
+        [kernel documentation](https://www.kernel.org/doc/html/v6.12-rc1/scheduler/sched-ext.html#switching-to-and-from-sched-ext) for details.
+        :::
+      '';
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.scx.full;
+      defaultText = lib.literalExpression "pkgs.scx.full";
+      example = lib.literalExpression "pkgs.scx.rustland";
+      description = ''
+        The SCX package to use. The default is the `scx.full` package, which includes all schedulers.
+        You may choose a minimal package like `pkgs.scx.rustland` if you only need a specific one.
+
+        ::: {.note}
+        Overriding this does not change the default scheduler, you need to set `scheduler` option for that.
+        :::
+      '';
+    };
+
+    scheduler = lib.mkOption {
+      type = lib.types.enum [
+        "scx_bpfland"
+        "scx_central"
+        "scx_flatcg"
+        "scx_lavd"
+        "scx_layered"
+        "scx_nest"
+        "scx_pair"
+        "scx_qmap"
+        "scx_rlfifo"
+        "scx_rustland"
+        "scx_rusty"
+        "scx_simple"
+        "scx_userland"
+      ];
+      default = "scx_rustland";
+      example = "scx_bpfland";
+      description = ''
+        Which of the SCX's schedulers to use. See the
+        [scx docs](https://github.com/sched-ext/scx/tree/main/scheds) for details
+        and which one is suitable for your needs.
+      '';
+    };
+
+    extraArgs = lib.mkOption {
+      type = lib.types.listOf lib.types.singleLineStr;
+      example = [
+        "--slice-us 5000"
+        "--verbose"
+      ];
+      description = ''
+        Options passed to the SCX scheduler at runtime.
+
+        ::: {.note}
+        See `preferred-scx-scheduler --help` to see the available options.
+        Generally, each scheduler has its own set of options. and they are
+        incompatible with each other.
+        :::
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
```

Comment:
You can assert on the kernel config here, see `requiredKernelFeatures`.

Reply:
I am not finding any result when I look up `requiredKernelFeatures` in across Nixpkgs, perhaps you mean `requiredKernelConfig`?

I suppose this would work?
```
    system.requiredKernelConfig = with config.lib.kernelConfig; [
      (isYes "SCHED_CLASS_EXT")
      (isYes "BPF")
    ];
```

Reply:
Yes, that.

Reply:
Done. Purposefully excluded the PAHOLE options.